packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -1210,42 +1229,53 @@ class MediaQuery extends InheritedModel<_MediaQueryAspect> {
     return InheritedModel.inheritFrom<MediaQuery>(context, aspect: aspect)?.data;
   }
 
-  /// Returns size for the nearest MediaQuery ancestor or
+  /// Returns [MediaQueryData.size] from the nearest [MediaQuery] ancestor or
   /// throws an exception, if no such ancestor exists.
   ///
+  /// {@template flutter.widgets.media_query.sizeOf.asyncNote}
+  /// This returns the size of the media in logical pixels (e.g, the size of the
+  /// screen) of a previously rendered frame.
+  ///
+  /// Because the information is delivered asynchronously, it is not
+  /// deterministic which previous frame supplied the size returned by [size].
+  /// The initial size is the size of the created [FlutterView].
+  /// {@endtemplate}
```

Comment:
I think we should remove the async note for now. This will become relevant in a multi window world again, but we haven't fully decided how to implement this yet. It is possible that we will fake the size to be correct.

For now, you can actually assume that `size` is the size the frame will be rendered at.